src/sentry/quotas/redis.py

```diff
@@ -34,19 +45,39 @@ class RedisQuota(Quota):
     grace = 60
 
     def __init__(self, **options):
-        self.cluster, options = get_cluster_from_options("SENTRY_QUOTA_OPTIONS", options)
+        self.is_redis_cluster, self.cluster, options = get_dynamic_cluster_from_options(
+            "SENTRY_QUOTA_OPTIONS", options
+        )
+
+        # NB: RedisCluster is a client, whereas RB needs to resolve one
+
         super(RedisQuota, self).__init__(**options)
         self.namespace = "quota"
 
     def validate(self):
         try:
-            with self.cluster.all() as client:
-                client.ping()
+            if self.is_redis_cluster:
+                self.cluster.ping()
+            else:
+                with self.cluster.all() as client:
+                    client.ping()
         except Exception as e:
             raise InvalidConfiguration(six.text_type(e))
 
-    def __get_redis_key(self, key, timestamp, interval, shift):
-        return u"{}:{}:{}".format(self.namespace, key, int((timestamp - shift) // interval))
+    def __get_redis_key(self, quota, timestamp, shift, organization_id):
+        interval = quota.window
+        if self.is_redis_cluster:
+            # new style redis cluster format which always has the organization id in
+            local_key = '%s{%s}%s' % (
+                quota.prefix,
+                self.organization_id,
```

Comment:
Where's `self.organization_id` coming from? Shouldn't this be the function parameter from line 67?